lib/client/client.js

```diff
@@ -110,12 +110,14 @@ function Client (options) {
   EventEmitter.call(this, options)
 
   var self = this
-  var _url
-  if (options.url) { _url = url.parse(options.url) }
-  this.host = _url ? _url.hostname : undefined
-  this.port = _url ? _url.port : false
-  this.secure = _url ? _url.secure : false
-  this.url = _url
+  this.urls = options.url ? [].concat(options.url).map(url.parse) : []
+  // select random server at the start
+  this._nextServer = Math.floor(Math.random() * this.urls.length)
```

Comment:
Should it be random? Or should we try to connect to each server sequentially? I can easily envision someone supplying an array of servers ordered in their preferred connect order.

Reply:
So for multiple server strategies we could have:

1. Redundancy random order
2. Redundancy prefered order
3. Lowest latency

Maybe we need a way for people to instruct ldapjs which one they want?

Reply:
That's a lot to ask for this PR and the codebase as it currently exists. 2 and 3 are easily met by the user determining viability before constructing the array to pass in (i.e. they ping the things and create the array in latency order).

If you two think random is the way to go, then we'll ship it as-is. Otherwise, the only change I'd make before merging is using the array sequentially as, in my opinion, it is going to be the case assumed by most users.

Reply:
Should this configuration be a part of `reconnect` option? Or should I create new option `connectStrategy`?

Reply:
I agree the `url` array could be randomized each time it is sent to ldapjs if someone wants it to be randomized.

We probably don't need other strategies, just the preferred order.